test/lib/redis-store-zlib-spec.js

```diff
@@ -0,0 +1,372 @@
+var config = require('../config.json');
+var redisStore = require('../../index');
+var sinon = require('sinon');
+var assert = require('assert');
+var zlib = require('zlib');
+
+var redisGzipCache;
+var customRedisGzipCache;
+var testJson;
+
+describe('Gzip Tests', function () {
+
+  before(function () {
+    redisGzipCache = require('cache-manager').caching({
+      store: redisStore,
+      host: config.redis.host,
+      port: config.redis.port,
+      auth_pass: config.redis.auth_pass,
+      db: config.redis.db,
+      ttl: config.redis.ttl,
+      gzip: true
```

Comment:
It's an interesting idea. It would would need to be modularized and the correct module would need to be lazy-loaded based on the options provided. It could wind up being a good-sized refactor, though.

You may want to find out how much interest there is in the community for other compression algorithms, first. I can think of a few -- snappy, LZ4, LZO, etc... -- but I don't know how many people understand the differences between them or would want to use them. The nice thing about `gzip` is that it's built into the Node `Zlib` module and has no further dependencies. It's not the fastest algorithm, but it's a decent tradeoff, especially when set to it's lowest compression / fastest speed configuration -- which is what I typically use.

Anyway, I am open to the idea. I always prefer my tools being more flexible, rather than less.  :)